go-kit/metrics/hllcounter.go

```diff
@@ -0,0 +1,76 @@
+/* Copyright (c) 2018 Salesforce
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package metrics
+
+import (
+	"sync"
+
+	hll "github.com/axiomhq/hyperloglog"
+)
+
+var (
+	_ CardinalityCounter = &HLLCounter{}
+)
+
+// HLLCounter provides a wrapper around a HyperLogLog probabalistic
+// counter, capable of being reported to Librato.
+type HLLCounter struct {
+	Name    string
+	lvs     []string
+	mu      sync.RWMutex
```

Comment:
Assuming `mu` will mostly be locked for `Insert()` calls, a plain old `sync.Mutex` might be more performant.